### ISSUE TITLE
Allow Regex for `ignoreAnnotated` on `UnusedPrivateMember`

### ISSUE THREAD

**Issue by wbrawner** [detekt/detekt#5537]
## Expected Behavior of the rule

Instead of a fixed string, using a regex would be nice. E.g.

```yaml
  UnusedPrivateMember:
    ignoreAnnotated: ['\w*Previews?']
```

## Context

The docs for [usage with Jetpack Compose]() state that one should set `ignoreAnnotated` to `['Preview']` to have detekt ignore composable previews. Jetpack Compose allows you to [create custom annotation classes](https://developer.android.com/jetpack/compose/tooling#preview-multipreview) to combine multiple `@Preview` annotations, but then each multipreview annotation class needs to be added to the ignored list. A regex would allow one to cover all of their various preview annotations.

**Comment by cortinico** [comment]
> but then each multipreview annotation class needs to be added to the ignored list

I would actually argue that having all the annotations listed will make the config more explicit and less error prone.

**Comment by wbrawner** [comment]
Fair enough! I'll close this issue for now unless I can come up with a more compelling reason than "it's slightly more convenient"
